pkg/build/buildkit/buildx.go

```diff
@@ -207,3 +219,28 @@ func (b *Builder) TagInvocationImage(ctx context.Context, origTag, newTag string
 	}
 	return nil
 }
+
+func convertMap(mapInput map[string]interface{}) (map[string]string, error) {
```

Comment:
We can define this in a new package, maybe pkg/build/driverutil, and then call it something a bit closer to what it is really doing, like "FlattenMap". Then you can call it from both places with `driverutil.FlattenMap(...)`